### DISCUSSION TITLE
Need help using RabbitMQ with SSL

### DISCUSSION THREAD

**Discussion by gpuliyar** [shlinkio/discussions#133] [Help wanted] [answered]
### Shlink version

3.10.2

### PHP version

Docker image

### How do you serve Shlink

RoadRunner Docker image

### Database engine

MySQL

### Database version

8.0.0

### Summary

I'm using AWS RabbitMQ, which comes with default SSL. I have configured the correct values in the environment, and the events are not getting published to MQ. When I locally test with RabbitMQ without SSL, it works as expected. I'm wondering if I need to be more accurate in the setup in the Cloud or if Shlink does not support RabbitMQ with SSL.

**Comment by gpuliyar** [comment]
I'm sharing more information about the Setup. I deployed `shlink` as an ECS Fargate Service. The app boots up correctly. It uses MySQL, Redis, and MQ. It can interact with RDS and Cache but not with MQ. I can see entries getting created in RDS and Cache. But nothing happens in MQ. The Security Groups are configured correctly. I don't see any meaningful errors showing in the logs if at all there is any failure encountered by the application while it connects with MQ. What is the best way to determine the root cause and fix the issue?

**Comment by gpuliyar** [comment]
I also wrote a separate consumer application that connects with the same RabbitMQ configured to Shlink - also deployed as an ECS Fargate Service. The consumer application can connect to the MQ broker and listen to the Queue. So, I know the MQ configurations are working as expected, and there is no trouble with the networking.

**Comment by gpuliyar** [comment]
I'm using the Container - `shlinkio/shlink:stable-non-root`

**Comment by gpuliyar** [comment]
@acelaya - could you let me know the answer to this question?

**Comment by acelaya** [comment, marked as answer]
EDIT 2023-11-25: Shlink 3.7.0 now supports encrypted RabbitMQ connections via TLS.

---

By taking a quick look at the options used by the AMQP abstraction library that Shlink uses, it seems there's some extra config needed in order to support connections over SSL.

With that considered, plus the fact that it was never tested, I'm almost sure it is not currently possible without changes in Shlink's implementation.

I've created an issue to track this: https://github.com/shlinkio/shlink/issues/1906

**Comment by gpuliyar** [comment]
Thank you, @acelaya, for your response and the created issue. 🙏🏾

**Reply by acelaya** [reply]
I have managed to make it work through SSL using a free-tier server from cloudamqp.com.

I verified it didn't work there with current implementation, because they also use encrypted connections, but I know AWS tends to be more picky, so I want to test the changes there as well. The problem is that it is a bit harder to set-up stuff (VPC, rabbitmq, EC2 server, etc), but this looks promising.

**Reply by acelaya** [reply]
I have just set-up the RabbitMQ instance on AWS and gladly seen they allow you to expose it to the internet... Right after I already had set-up the VPC, EC2 instance, etc 😅 

But anyway, the good news is that I could verify it now works with AWS as well. I'm releasing v3.7.0 in a couple of hours which will include support for encrypted RabbitMQ connections by providing the new `RABBITMQ_USE_SSL=true` env var.
